src/utils/stacktrace.ts

```diff
@@ -0,0 +1,95 @@
+import ErrorStackParser from "error-stack-parser";
+
+export type RawStackFrames = string;
+
+const getErrorTitle = (e: Error): string => `${e.name || "Error"}: ${e.message}`;
+const getErrorStackFrames = (e: Error): string | undefined => e.stack?.replace(getErrorTitle(e) + "\n", "");
+
+export function captureRawStackFrames(filterFunc?: (...args: unknown[]) => unknown): RawStackFrames {
+    const savedStackTraceLimit = Error.stackTraceLimit;
+    const targetObj = {} as { stack: RawStackFrames };
+
+    Error.stackTraceLimit = 64;
+    Error.captureStackTrace(targetObj, filterFunc || captureRawStackFrames);
+    Error.stackTraceLimit = savedStackTraceLimit;
+
+    const rawFramesPosition = targetObj.stack.indexOf("\n") + 1; // crop out error message
+
+    return targetObj.stack.slice(rawFramesPosition);
+}
+
+export function applyStackFrames(error: Error, frames: RawStackFrames): Error {
+    if (!error || !error.message) {
+        return error;
+    }
+
+    error.stack = getErrorTitle(error) + "\n" + frames;
+
+    return error;
+}
+
+export function filterExtraWdioFrames(error: Error): Error {
+    if (!error || !error.message || !error.stack) {
+        return error;
+    }
+
+    const frames = getErrorStackFrames(error)!;
```

Comment:
Why doesn't `getErrorStackFrames` method  return array with frames? By its name, I would expect this behavior.

Reply:
Because it is located near the `captureRawStackFrames` and `captureRawStackFrames` returns string. Replaced it with:
```ts
type ErrorWithStack = SetRequired<Error, "stack">;

const getErrorRawStackFrames = (e: ErrorWithStack): RawStackFrames => e.stack.replace(getErrorTitle(e) + "\n", "");
```
so it would be more clear